pkg/generate/git/git.go

```diff
@@ -21,13 +21,14 @@ import (
 // - ssh
 func ParseRepository(s string) (*url.URL, error) {
 	uri, err := url.Parse(s)
-	if err != nil {
-		return nil, err
+	if err == nil && uri.Scheme != "" {
```

Comment:
What kind of change it should be? For Go 1.8 it already covered by current tests, for earlier versions of Go it works fine with and without changes.

```
$ go version
go version go1.8 darwin/amd64
$ go test github.com/openshift/origin/pkg/generate/git
--- FAIL: TestParseRepository (0.06s)
	git_test.go:101: ParseRepository returned err: parse mbalazs@build.ulx.hu:/var/git/eap-ulx.git: first path segment in URL cannot contain colon
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x125881e]
```